addons/html_editor/static/src/others/embedded_components/plugins/toggle_plugin/toggle_plugin.js

```diff
@@ -0,0 +1,627 @@
+import { getEmbeddedProps } from "@html_editor/others/embedded_component_utils";
+import { Plugin } from "@html_editor/plugin";
+import { baseContainerGlobalSelector } from "@html_editor/utils/base_container";
+import { closestBlock } from "@html_editor/utils/blocks";
+import { removeClass } from "@html_editor/utils/dom";
+import { isEmptyBlock, isParagraphRelatedElement } from "@html_editor/utils/dom_info";
+import {
+    childNodes,
+    children,
+    closestElement,
+    firstLeaf,
+    lastLeaf,
+    selectElements,
+} from "@html_editor/utils/dom_traversal";
+import { parseHTML } from "@html_editor/utils/html";
+import { childNodeIndex, nodeSize } from "@html_editor/utils/position";
+import { withSequence } from "@html_editor/utils/resource";
+import { _t } from "@web/core/l10n/translation";
+import { renderToString } from "@web/core/utils/render";
+import { uuid } from "@web/views/utils";
+
+const toggleSelector = "[data-embedded='toggle']";
+const titleSelector = "[data-embedded-editable='title']";
+const contentSelector = "[data-embedded-editable='content']";
+
+export class TogglePlugin extends Plugin {
+    static id = "toggle_list";
+    static dependencies = [
+        "baseContainer",
+        "delete",
+        "dom",
+        "embeddedComponents", // toggle is an embedded component.
+        "hint", // needed to clean custom toggle hints.
+        "history",
+        "selection",
+        "split",
+    ];
+    resources = {
+        hints: [
+            withSequence(1, {
+                selector: `${toggleSelector} ${titleSelector} > *`,
+                text: _t("Toggle title"),
+            }),
+        ],
+        move_node_selectors_blacklist: `${toggleSelector} ${titleSelector} *`,
+        powerbox_items: [
+            {
+                commandId: "insertToggleList",
```

Comment:
Obviously I'm 😬 at every mention of "toggleList" that means something else than the `toggleList` command... ^^' I'm mentioning this one just to make my point accross, but obviously it applies at lots of other places too... ^^'